js/charts/clusteredverticalbar.js

```diff
@@ -878,78 +875,76 @@ function clusteredverticalbar() {
+                                broadcast.$broadcast('FlairBi:livemode-dialog');
+                                return;
                             }
-                            else {
-                                if (isLiveEnabled) {
-                                    broadcast.$broadcast('FlairBi:livemode-dialog');
-                                    return;
-                                }
 
-                                filter = false;
-                                var confirm = parentContainer.select('.confirm')
-                                    .style('visibility', 'visible');
-                                var _filter = _Local_data.filter(function (d1) {
-                                    return d[_dimension[0]] === d1[_dimension[0]]
+                            filter = false;
+                            var confirm = parentContainer.select('.confirm')
+                                .style('visibility', 'visible');
+                            var _filter = _Local_data.filter(function (d1) {
+                                return d[_dimension[0]] === d1[_dimension[0]]
+                            })
+                            var rect = d3.select(this);
+                            if (rect.classed('selected')) {
+                                rect.classed('selected', false);
+                                filterData.map(function (val, i) {
+                                    if (val[_dimension[0]] == d[_dimension[0]]) {
+                                        filterData.splice(i, 1)
+                                    }
                                 })
-                                var rect = d3.select(this);
-                                if (rect.classed('selected')) {
-                                    rect.classed('selected', false);
-                                    filterData.map(function (val, i) {
-                                        if (val[_dimension[0]] == d[_dimension[0]]) {
-                                            filterData.splice(i, 1)
-                                        }
-                                    })
-                                } else {
-                                    rect.classed('selected', true);
-                                    var isExist = filterData.filter(function (val) {
-                                        if (val[_dimension[0]] == d[_dimension[0]]) {
-                                            return val
-                                        }
-                                    })
-                                    if (isExist.length == 0) {
-                                        filterData.push(_filter[0]);
+                            } else {
+                                rect.classed('selected', true);
+                                var isExist = filterData.filter(function (val) {
+                                    if (val[_dimension[0]] == d[_dimension[0]]) {
```

Comment:
what is the diff b/w 2 objects ...val[_dimension[0]]  d[_dimension[0]..can u give some meaningful names so that we can indetify them

Reply:
d and Val are coding standards in d3 mostly use in while setting attribute or vales like "i" in loops